### PR TITLE
Remove a redundant recursive call to `gtSetEvalOrder`.

### PR DESCRIPTION
**Pull request by pgavlin**

### CODE REVIEW

File: src/jit/gentree.cpp [hunk 1]
```diff
@@ -3333,8 +3333,6 @@ unsigned            Compiler::gtSetEvalOrder(GenTree * tree)
 
             case GT_MKREFANY:
             case GT_OBJ:
-                level  = gtSetEvalOrder(tree->gtOp.gtOp1);
-                ftreg |= tree->gtOp.gtOp1->gtRsvdRegs;
```
Comment (pgavlin) [inline src/jit/gentree.cpp]: These operations are already performed above: https://github.com/dotnet/coreclr/pull/5277/files#diff-26597eae27417b7113176d9fcb5e1f10L3144

**Comment by pgavlin** [comment]
@briansull @CarolEidt @dotnet/jit-contrib PTAL

**Comment by pgavlin** [comment]
I have verified that there are no diffs when crossgen'ing mscorlib with this change.

File: src/jit/gentree.cpp (hunk 1 above)
Comment (briansull) [inline src/jit/gentree.cpp]: What about this same sequence below on GT_BOX?
Isn't that also redundant?

Comment (pgavlin) [inline src/jit/gentree.cpp]: It is, yes. I'll remove it.

**Comment by briansull** [comment]
LGTM with commenst

**Comment by sivarv** [comment]
Looks good.

**Comment by pgavlin** [comment]
@dotnet-bot test Windows_NT x64 Release Priority 1 Build and Test

**Comment by pgavlin** [comment]
@dotnet-bot test Windows_NT x64 Release Priority 1 Build and Test

**Comment by CarolEidt** [comment]
LGTM

**Comment by pgavlin** [comment]
test Windows_NT x64 Release Priority 1 Build and Test

**Pull request merged**